mantidproject/mantid
# Set last run to "Auto" in ALC interface

Resolves #10057.

User can enter the last run as "auto", which will automatically set the last run to the most recently modified file in the same directory as first_run.
The interface will then monitor this directory and automatically add files to the data set as they are added to the directory (or removed). 

Test:
- Get a set of muon data, for example HIFI 83077-83157 (\isis\inst$\cycle_15_1\NDXHIFI...) and put a subset of these files in a test directory.
- Open ALC interface, set first_run to the first file in the test directory.
- Set last run to "Auto" (not case-sensitive) and click Load. Verify that all files in the directory are automatically loaded.
- Copy some more of the data sets into the test directory. Verify that they are automatically picked up and loaded.
- Test that interface can handle some sets being removed from the directory.
- Test that existing functionality is not broken, i.e. setting last_run to a file (not auto) works.

Could not add a unit test for this because the data files used in the tests do not appear in a directory, so cannot find the other files automatically. This did however prompt the handling of exceptions that may be thrown in this situation.

Documentation updates:
- [x] Update [release notes](http://www.mantidproject.org/Release_Notes_3_6_MuonAnalysis)
- [x] Update [interface documentation](http://www.mantidproject.org/Muon_ALC:_Data_Loading)
